demisto_sdk/commands/format/update_generic_yml.py

```diff
@@ -143,7 +147,29 @@ def update_tests(self) -> None:
                             test_playbook_data = get_yaml(tpb_file_path)
                             test_playbook_id = get_entity_id_by_entity_type(test_playbook_data,
                                                                             content_entity='')
-                            test_playbook_ids.append(test_playbook_id)
+                            if not scripts and not commands:  # Better safe than sorry
+                                test_playbook_ids.append(test_playbook_id)
+                            else:
+                                added = False
+                                tpb_commands, tpb_scripts = get_scripts_and_commands_from_yml_data(
+                                    test_playbook_data, FileType.TEST_PLAYBOOK)
+
+                                for full_tpb_command in tpb_commands:
+                                    tpb_command_parts = full_tpb_command.split('|||')
+                                    tpb_command = full_tpb_command
+                                    if len(tpb_command_parts) == 2:
+                                        if file_id and file_id != tpb_command_parts[0]:
+                                            continue
+                                        tpb_command = tpb_command_parts[1]
+
+                                    if not added and tpb_command in commands:
+                                        test_playbook_ids.append(test_playbook_id)
+                                        added = True
+                                for tpb_script in tpb_scripts:
+                                    if not added and tpb_script in scripts:
+                                        test_playbook_ids.append(test_playbook_id)
+                                        added = True
```

Comment:
With the split logic removed to the tools func
some simplifying (IMO, if you think this is worse - welcome to not commit suggestion)
- perhaps move the any or any if cond to a small func if its not readable)
```suggestion

                                # Filter commands with other brands
                                tpb_commands = [tpb_command for tpb_command in tpb_commands if not tpb_command.get('brand') or tpb_command.get('brand') == file_id]
                                if any(tpb_command in commands for tpb_command in tpb_commands) or any(tpb_script in scripts for tpb_script in tpb_scripts):
                                    test_playbook_ids.append(test_playbook_id)
```

Reply:
I tried to implement this but it misses the case where the command inside the integration doesn't have a prefix while in the testPlaybook it does, so I left the granularity needed as is. I handled the split in the tools method as you suggested.